src/EnergyPlus/FileSystem.cc

```diff
@@ -338,100 +346,89 @@ namespace FileSystem {
 
     std::string readFile(fs::path const &filePath, std::ios_base::openmode mode)
     {
-#ifdef _WIN32
-        std::string filePathStr = filePath.string();
-        const char *path = filePathStr.c_str();
-#else
-        const char *path = filePath.c_str();
-#endif
-
+        // Shenanigans would not be needed with fmt 10+ (maybe earlier), because fmt has native fs::path support
         if (!fileExists(filePath)) {
-            throw FatalError(fmt::format("File does not exists: {}", path));
+            throw FatalError(fmt::format("File does not exists: {}", filePath));
         }
 
-        std::string_view fopen_mode;
-        if (mode == std::ios_base::in) {
-            fopen_mode = "r";
-        } else if (mode == std::ios_base::binary) {
-            fopen_mode = "b";
-        } else if (mode == (std::ios_base::in | std::ios_base::binary)) {
-            fopen_mode = "rb";
-        } else {
+        // Can only be 'r', 'b' or 'rb'
+        if ((mode & (std::ios_base::in | std::ios_base::binary)) == 0) {
             throw FatalError("ERROR - readFile: Bad openmode argument. Must be std::ios_base::in or std::ios_base::binary");
         }
 
-        auto close_file = [](FILE *f) { fclose(f); };
-        auto holder = std::unique_ptr<FILE, decltype(close_file)>(fopen(path, fopen_mode.data()), close_file); // (THIS_AUTO_OK)
-        if (!holder) {
-            throw FatalError(fmt::format("Could not open file: {}", path));
-        }
-
-        auto f = holder.get(); // (THIS_AUTO_OK)
-        const std::uintmax_t size = fs::file_size(filePath);
-        std::string result;
-        result.resize(size);
-
-        size_t bytes_read = fread(result.data(), 1, size, f);
-        bool is_eof = feof(f);
-        bool has_error = ferror(f);
-        if (is_eof != 0) {
-            return result;
-        }
-        if (has_error != 0 || bytes_read != size) {
-            throw FatalError(fmt::format("Error reading file: {}", path));
+        const std::uintmax_t file_size = fs::file_size(filePath);
+        std::ifstream file(filePath, mode);
+        if (!file.is_open()) {
+            throw FatalError(fmt::format("Could not open file: {}", filePath));
         }
+        std::string result(file_size, '\0');
+        file.read(result.data(), file_size);
```

Comment:
Change readFile and readJSON to use ifstream (that handles fs::path) + avoid using path.string()